src/inventory.cpp

```diff
@@ -652,8 +652,8 @@ void _cb_item_recv_RequestWeaponReload_actionUpdate(mapChannel_t* mapChannel, ac
 			if (itemAmmo->itemTemplate->item.classId == ammoClassId)
 			{
 				// consume ammo
-				sint32 ammoToGrab = min(item->itemTemplate->weapon.clipSize - foundAmmoAmount, itemAmmo->stacksize);
-				foundAmmoAmount += ammoToGrab;
+				sint32 ammoToGrab = min(item->itemTemplate->weapon.clipSize - foundAmmoAmount - item->weaponData.ammoCount, itemAmmo->stacksize);
+				foundAmmoAmount = ammoToGrab + item->weaponData.ammoCount;
 				inventory_reduceStackCount(client, itemAmmo, ammoToGrab);
```

Comment:
Fixed bug where on reload weapon, every time consumed max clip size of ammo.
Now only use how much iss missing.